lib/mrtrix3/file.py

```diff
@@ -73,54 +73,69 @@ def newTempFile(suffix):
 # Wait until a particular file not only exists, but also does not have any
 #   other process operating on it (so hopefully whatever created it has
 #   finished its work)
-# Using open() rather than os.path.exists() ensures that not only has the file
-#   appeared in the directory listing, but the data are there and a file handle
-#   can be obtained. Additionally, an exclusive file lock is obtained.
-# 'r+' mode is used since it includes write access (which may fail if some other
-#   process also has write access to that file), but will _not_ create a new file
-#   if the file is somehow deleted between os.path.exists() and open().
+# This functionality is achieved in different ways, depending on the capabilities
+#   of the system:
+#   - On Windows, two processes cannot open the same file in read mode. Therefore,
+#     try to open the file in 'rb+' mode, which requests write access but does not
+#     create a new file if none exists
+#   - If command fuser is available, parse its output and look for any processes
+#     that have write access to the file.
+#   - If neither of those applies, no additional safety check can be performed.
 # Initially, checks for the file once every 1/1000th of a second; this gradually
 #   increases if the file still doesn't exist, until the program is only checking
 #   for the file once a minute.
 def waitFor(path):
   import os, time
-  
-  try:
-    import fcntl
-    do_fcntl = True
-  except ImportError:
-    do_fcntl = False
   from mrtrix3 import app
+
+  def inUse(path):
+    import subprocess
+    from distutils.spawn import find_executable
+    if app.isWindows():
+      if not os.access(path, os.W_OK):
+        return None
+      try:
+        with open(path, 'rb+') as f:
+          pass
+        return False
+      except:
+        return True
+    if not find_executable('fuser'):
+      return None
+    # Apparently the 'F' for open with write access only appears in the verbose output
```

Comment:
I'm not sure I'm comfortable looking through the verbose output, there's no guarantee that it won't change. I think it would be safer to rely purely on the exit code. You've already checked that the file exists and is accessible (assuming you go with my previous suggestion of running the `os.access()` check regardless of OS), so that ambiguity is not an issue. Checking the verbose output allows you to test whether the file is opened with write access, but I'm not sure that's necessarily a requirement. You don't expect the file to be opened read-only either, and if another process was accessing it read-only, you'd probably need to wonder why, right...?